pilot/pkg/controllers/ipallocate/ipallocate.go

```diff
@@ -233,7 +240,7 @@ func (c *IPAllocator) resolveConflict(conflict conflictDetectedEvent) error {
 			continue
 		}
 		log.Warnf("reassigned %s/%s due to IP Address conflict", resolveMe.Namespace, resolveMe.Name)
-		patch, err := c.statusPatchForAddresses(resolveMe, true)
+		patch, _, err := c.statusPatchForAddresses(resolveMe, true)
```

Comment:
do we need to handle both here? The conflict could be from a `spec.address` I think, right?

Reply:
My thought was if we have a conflict it means we have a status with addresses that need changing, maybe that's not always true though 🤔

Reply:
ok, looking over the code again I think its correct but the conversation highlights that its not so clear, I should add a clarifying comment